code/modules/admin/verbs/adminpanelweapons.dm

```diff
@@ -2,120 +2,56 @@
 	set name = "Weapons"
 	set category = "Admin.Ship"
 
-	var/weapontype = tgui_alert(src, "What weapon?", "Choose wisely!", list("Missile", "Railgun"), 20 SECONDS)
-	if(!weapontype)
+	var/datum/space_weapon/weapon_type = tgui_alert(src, "What weapon?", "Choose wisely!", GLOB.space_weapons, 20 SECONDS)
+	if(!weapon_type)
 		return
-	var/hiteta = tgui_input_number(src, "Give an ETA for the weapon to hit.", "Don't make them wait too long!", 10, 120, 10, 20 SECONDS)
-	if(!hiteta)
-		return
-	var/point_defense = tgui_alert(src, "Allow Point Defence of the ship to intercept, or for the weapon to miss?", "standard  PD/miss chance is 30%.", list("Yes", "No"), 20 SECONDS)
-	if(!point_defense)
-		return
-	point_defense = point_defense == "Yes"
-	var/exactplace = tgui_alert(src, "Shoot it at random places, or where you're at?", "Choose wisely!", list("Random", "Where I am"), 20 SECONDS)
-	if(!exactplace)
-		return
-	exactplace = exactplace == "Where I am"
-
-	var/salvo
-	var/quantity
-	if(exactplace == FALSE)
-		salvo = tgui_alert(src, "Make it a salvo or a single fire?", "Choose wisely!", list("Salvo", "Single"), 20 SECONDS)
-		if(!salvo)
-			return
-		salvo = salvo == "Salvo"
-		if(salvo == TRUE)
-			quantity = tgui_input_number(src, "How many?", "Don't go overboard. Please.", 2, 10, 2, 20 SECONDS)
+	weapon_type = GLOB.space_weapons[weapon_type]
 
-	var/prompt = tgui_alert(src, "Are you sure you want to open fire at the USS Almayer with those parameters?", "Choose wisely!", list("Yes", "No"), 20 SECONDS)
-	if(prompt != "Yes")
+	var/list/ammo_type = list()
+	if(tgui_alert(src, "Supposed to use all ammo types?", "Ammo selector", list("Yes", "No")) == "Yes") 
+		ammo_type = weapon_type.possibly_ammunition
+	else
+		var/list/potential_ammo_list = weapon_type.possibly_ammunition
+		var/number_of_ammo = tgui_input_number(src, "How many?", "Ammo selector", 1, length(potential_ammo_list) - 1, 1, 20 SECONDS)
+		for(var/i = 1 to number_of_ammo)
+			var/additional_ammo = tgui_alert(src, "Choose ammo", "Ammo selector", potential_ammo_list, 20 SECONDS)
```

Comment:
It would be better to just have the loop asking to add a new ammo type (if any remain to add), removing from the possible options the previous option added. The admin won't necessarily know what all ammo types there are to choose from the first place to make the decision how much variety they should have.

In other words:
While possible ammo list > 0: Add which ammo? > Save the choice, and remove it from the possible list.